server/datastore/redis/redis.go

```diff
@@ -260,18 +264,36 @@ func PublishHasListeners(pool fleet.RedisPool, conn redis.Conn, channel, message
 }
 
 func newCluster(conf PoolConfig) (*redisc.Cluster, error) {
+	// Initialize AWS IAM token generator if needed
+	var awsIAMTokenGen *awsIAMAuthTokenGenerator
+
 	opts := []redis.DialOption{
 		redis.DialDatabase(conf.Database),
 		redis.DialUseTLS(conf.UseTLS),
 		redis.DialConnectTimeout(conf.ConnTimeout),
 		redis.DialKeepAlive(conf.KeepAlive),
 		redis.DialUsername(conf.Username),
-		redis.DialPassword(conf.Password),
 		redis.DialWriteTimeout(conf.WriteTimeout),
 		redis.DialReadTimeout(conf.ReadTimeout),
 	}
 
+	// Auto-detect ElastiCache and use IAM auth if no password is provided
+	useIAMAuth := false
+	if conf.Password == "" && conf.Region != "" && conf.CacheName != "" {
```

Comment:
I think the region should be optional, in the vast majority of cases the region will just be whatever the default is from the environment, and here we don't need it because `CacheName` can be used.